### ISSUE TITLE
🐞 Console shows missing CSS Stylesheet for the Loop Item template - [ED-13534]

### ISSUE THREAD

**Issue by samuelmente** [elementor/elementor#24959]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

Since the Update to Elementor Pro 3.19.0, for loops 2 css-Sheets are included for each loop:

1. loop-<post-id>.css (example: loop-123.css)
2. post-<post-id>.css (example: loop-123.css)

loop-123.css is properly loaded while post-123.css can't be found (404).

I think that you added a change with the update to include the css for loops as "loop-id.css" instead of "post-id.css" to better differentiate them, but somewhere in the code probably forgot to remove the call for the old naming scheme.

### Steps to reproduce

1. Update to Elementor Pro 3.19.0
2. Go to a page with a loop-grid
3. Go into the Dev-Tools and see that a stylesheet can't be loaded, because it can't be found

### Expected behavior

1. Update to Elementor Pro 3.19.0
2. Go to a page with a loop-grid
3. Go into the Dev-Tools and see that there are no errors

### Isolating the problem

- [X] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [X] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt
== Server Environment ==
	Operating System: Linux
	Software: nginx/1.24.0
	MySQL version: mariadb.org binary distribution v10.11.6-MariaDB-1:10.11.6+maria~ubu2204
	PHP Version: 8.0.30
	PHP Memory Limit: 512M
	PHP Max Input Vars: 10000
	PHP Max Post Size: 2000M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.4.2
	Site URL: **redacted**
	Home URL: **redacted**
	WP Multisite: No
	Max Upload Size: 2 GB
	Memory limit: 512M
	Max Memory limit: 512M
	Permalink Structure: /%category%/%postname%/
	Language: de-DE
	Timezone: Europe/Berlin
	Debug Mode: Inactive

== Theme ==
	Name: Custom Hello Elementor Child Theme
	Version: 1.0.2
	Author: Samuel Mente
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 3.0.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: de_DE
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36

== Active Plugins ==
	ACF: Better Search
		Version: 4.2.0
		Author: Mateusz Gbiorczyk

	ACF Content Analysis for Yoast SEO
		Version: 3.1
		Author: Thomas Kräftner, ViktorFroberg, marol87, pekz0r, angrycreative, Team Yoast

	Advanced Custom Fields Multilingual
		Version: 2.0.5
		Author: OnTheGoSystems

	Advanced Custom Fields PRO
		Version: 6.2.5
		Author: WP Engine

	DynamicConditions
		Version: 1.6.2
		Author: RTO GmbH

	Elementor
		Version: 3.19.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.19.0
		Author: Elementor.com

	FacetWP
		Version: 4.2.11
		Author: FacetWP, LLC

	FacetWP - A-Z Listing
		Version: 1.3.9
		Author: FacetWP, LLC

	FacetWP - Cache
		Version: 1.7
		Author: FacetWP, LLC

	FacetWP - Elementor
		Version: 1.9
		Author: FacetWP, LLC

	FacetWP - Multilingual support
		Version: 1.0.1
		Author: FacetWP, LLC

	FacetWP Schedule Indexer
		Version: 1.1
		Author: FacetWP, LLC

	Local Google Fonts
		Version: 0.22.0
		Author: EverPress

	MainWP Child
		Version: 4.6
		Author: MainWP

	Media Cleaner (Pro)
		Version: 6.7.0
		Author: Jordy Meow

	Redirection
		Version: 5.4.2
		Author: John Godley

	Simple Image Slider
		Version: 1.0.0
		Author: Cakewp

	Sticky Posts - Switch
		Version: 2.1.3
		Author: Markus Wiesenhofer

	Transients Manager
		Version: 2.0.5
		Author: WPBeginner

	WebP Express
		Version: 0.25.8
		Author: Bjørn Rosell

	WP All Import - ACF Add-On
		Version: 3.3.8
		Author: Soflyy

	WP All Import Pro
		Version: 4.8.5
		Author: Soflyy

	WP Crontrol
		Version: 1.16.1
		Author: John Blackbourn & crontributors

	WP Mail SMTP Pro
		Version: 3.10.1
		Author: WP Mail SMTP

	WPML All Import
		Version: 2.3.0
		Author: OnTheGoSystems

	WPML Multilingual CMS
		Version: 4.6.9
		Author: OnTheGoSystems

	WPML SEO
		Version: 2.1.0
		Author: OnTheGoSystems

	WPML String Translation
		Version: 3.2.9
		Author: OnTheGoSystems

	Yoast Duplicate Post
		Version: 4.5
		Author: Enrico Battocchi & Team Yoast

	Yoast SEO
		Version: 21.9.1
		Author: Team Yoast

	Yoast SEO Premium
		Version: 21.7
		Author: Team Yoast


== Eigenschaften ==
	Custom Fonts: 1
	Custom Icons: 0

== Integrations ==
	
	recaptcha_v3: Aktiv


== Elementor-Experimente ==
	Verbessertes Laden von Assets: Aktiv
	Verbessertes Laden von CSS: Aktiv
	Inline-Schriftarten-Symbole: Inaktiv
	Zusätzliche benutzerdefinierte Breakpoints: Aktiv
	admin_menu_rearrangement: Standardmäßig deaktiviert
	Flexbox Container: Aktiv
	Upgrade Swiper Library: Aktiv
	Grid Container: Aktiv
	Default to New Theme Builder: Inaktiv
	Hello Theme Header & Footer: Inaktiv
	Editor Top Bar: Standardmäßig deaktiviert
	Optimized Gutenberg Loading: Standardmäßig aktiviert
	Build with AI: Standardmäßig aktiviert
	Startseiten: Aktiv
	Nested Elements: Aktiv
	Hintergrundbilder nachladen: Standardmäßig deaktiviert
	Optimize Image Loading: Standardmäßig aktiviert
	Globaler Styleguide: Standardmäßig aktiviert
	Page Transitions: Inaktiv
	Notes: Inaktiv
	Bedingungen anzeigen: Standardmäßig deaktiviert
	Form Submissions: Aktiv
	Scroll Snap: Inaktiv
	Menü: Standardmäßig deaktiviert
	Taxonomy Filter: Standardmäßig deaktiviert


== Protokoll ==
	
PHP: showing 4 of 4PHP: 2023-12-06 14:25:55 [warning X 1][../wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-text.php::33] Undefined array key 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-12-13 12:33:48 [warning X 16][../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined array key "condition_type" [array (
  'trace' => '
#0: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: ../wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2024-01-17 14:36:19 [warning X 2][../wp-content/plugins/elementor-pro/modules/posts/skins/skin-content-base.php::265] Attempt to read property "ID" on null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-01-24 13:47:10 [warning X 1][../wp-content/plugins/elementor/modules/history/revisions-manager.php::363] Undefined array key "editor_post_id" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

Log: showing 20 of 342023-12-27 17:03:31 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.18.1',
  'to' => '3.18.2',
)]
2023-12-27 17:05:04 [info] elementor::elementor_updater Started 
2023-12-27 17:05:04 [info] Elementor/Upgrades - _on_each_version Start  
2023-12-27 17:05:04 [info] Elementor/Upgrades - _on_each_version Finished 
2023-12-27 17:05:04 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.18.2',
  'to' => '3.18.3',
)]
2023-12-27 17:05:05 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.18.2',
  'to' => '3.18.3',
)]
2024-01-23 09:38:33 [info] elementor-pro::elementor_pro_updater Started 
2024-01-23 09:38:33 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-01-23 09:38:33 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-01-23 09:38:33 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.18.2',
  'to' => '3.18.3',
)]
2024-01-29 14:44:04 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.18.3',
  'to' => '3.19.0',
)]
2024-01-29 14:44:08 [info] elementor-pro::elementor_pro_updater Started 
2024-01-29 14:44:08 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-01-29 14:44:08 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-01-29 14:44:08 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.18.3',
  'to' => '3.19.0',
)]
2024-01-29 14:44:10 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.18.3',
  'to' => '3.19.0',
)]
2024-01-29 14:53:28 [info] elementor::elementor_updater Started 
2024-01-29 14:53:28 [info] Elementor/Upgrades - _on_each_version Start  
2024-01-29 14:53:28 [info] Elementor/Upgrades - _on_each_version Finished 
2024-01-29 14:53:28 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.18.3',
  'to' => '3.19.0',
)]

JS: showing 5 of 5JS: 2024-01-16 14:18:30 [error X 1][../wp-content/plugins/elementor/assets/js/4fdaa70e951ad90db2f2.bundle.min.js:2:7786] Cannot read properties of null (reading 'dataset') 
JS: 2024-01-16 14:21:42 [error X 1][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.18.3:3:695662] Cannot read properties of undefined (reading 'get') 
JS: 2024-01-23 12:42:32 [error X 3][../wp-content/plugins/elementor/assets/js/4fdaa70e951ad90db2f2.bundle.min.js:2:7998] Cannot read properties of undefined (reading 'view') 
JS: 2024-01-25 11:25:24 [error X 1][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.18.3:3:690111] this.model.isValidChild is not a function 
JS: 2024-01-26 11:34:51 [error X 1][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.18.3:3:640700] Cannot read properties of undefined (reading 'isDesignable') 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Die Kompatibilität ist nicht angegeben
	FacetWP - Elementor: Die Kompatibilität ist nicht angegeben

== Elementor Pro - Compatibility Tag ==

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by nicholaszein** [comment]
Hi @samuelmente, thank you for sharing your feedback! 🙏 


Please Regenerate Elementor's CSS & Data in your WordPress Dashboard > **`Elementor`** > **`Tools`** > **`General`** page, and clear/purge all levels of cache, including but not limited to **server-side** cache (static and **object cache**), **website** cache, **CDN** cache, and **browser** cache, and let me know if the issue is solved.

Kind regards

**Comment by samuelmente** [comment]
Hi @nicholaszein 
I already tried these steps on 2 completely different websites before submitting the bug. I tried regenerating CSS and purged all levels of cache. Did the same thing once again just now with the same result: the issue is still there.

**Comment by nicholaszein** [comment]
Thank you for confirming this @samuelmente! 🙏 

I've escalated this to our QA Team and we will work on a fix if validated.

Best regards!

**Comment by stefanbiba** [comment]
I'm having exactly the same issues with the latest version 3.19.0 on all my websites, which were working fine before updating. A rollback to 3.18.3. fixed the issue as a workaround.

**Comment by giannisgx89** [comment]
+1

**Comment by nicholaszein** [comment]
Hello everyone, just an update:

While the console will show there are missing styles for Loop Item templates, these are still being loaded but with another file name. So this shouldn't affect anything on your websites and you can update to 3.19.0 securely without any worries.

A fix is on it's way! Thank you for sharing your feedback! 🙏

**Comment by stefanbiba** [comment]
@nicholaszein Missing css files isn't the only issue as the performance with the latest update is so bad, that all pages with loop or slide items are getting really slow. I doubt the latest Wordpress core update will fix that.

**Comment by samuelmente** [comment]
@nicholaszein It is definitely not only the issue of it being displayed in the console. It also directly affects performance, as the page will wait to continue running js and loading some elements until the stylesheet is loaded (or in this case until the server returns 404) which in case of my test pages can take >1 second. Following is a screenshot for reference. During that "waiting" time all Entrance Animations for example will not run potentially adding a long delay until elements are visible.

![Timing](https://github.com/elementor/elementor/assets/58186629/f9bc60c5-5104-4f43-ba89-ac7950080a52)

**Comment by sermalefico** [comment]
in my case i lost 4s big bug

**Comment by KingGrizzly** [comment]
I can confirm the 404 loop error on multiple sites. Seems to occur in 3.18.3 as well.

**Comment by SKOR-DEV** [comment]
I encountered the same issue.

<img width="889" alt="404 loop error" src="https://github.com/elementor/elementor/assets/110029190/5b40aadd-3bb7-43d2-a7ca-7d4ab52bb010">

The thing is, if you don't add custom CSS to a loop item, this file is not really necessary. In my case, I simply disabled its loading through the Perfmatters Script Manager, and it worked for me. If additional CSS is needed, you can create your own file and load it where necessary.

<img width="870" alt="perfmatters script manager" src="https://github.com/elementor/elementor/assets/110029190/ccce1067-ac60-4735-9a19-d9622faca39e">

Another solution that worked for me was to use a plugin that allows setting a complete page template as a loop item. However, this was not as efficient and required an additional plugin.

A third option I considered but did not try involves manually creating a file at the missing URL. I'm not sure if this would work or what would happen after an Elementor update in such a case.

A fourth option is:
1. Go to Elementor > Settings > Advanced.
2. Find the CSS Print Method option and change it to Internal Embedding.

However, doing this means your CSS will not be loaded as separate files, and therefore they will not be cached. This could impact performance.

Everything needs to be checked individually. But if the error is not fixed, the site's speed significantly decreases due to the prolonged search for this style.

**Comment by jamieburchell** [comment]
> Hello everyone, just an update:
> 
> While the console will show there are missing styles for Loop Item templates, these are still being loaded but with another file name. So this shouldn't affect anything on your websites and you can update to 3.19.0 securely without any worries.
> 
> A fix is on it's way! Thank you for sharing your feedback! 🙏

When? This is pretty bad as it's slowing down the rendering of pages.

**Comment by samuelmente** [comment]
As of today with the release of Elementor Pro 3.19.1 it seems to be fixed.

**Comment by jamieburchell** [comment]
Is nobody else concerned that this wasn't picked up in testing and then took a week to fix with a few lines of code? I mean bugs happen, but it literally took a glance at either the browser console or network tab on a vanilla Elementor/pro install which uses a loop template.

**Comment by SKOR-DEV** [comment]
For me, the Elementor Pro 3.19.1 update solved the problem. Thank you for your prompt work!

**Comment by lachermeierm** [comment]
I use the "Dynamic Template Grid" widget from "Unlimited Elements for Elementor" in the Pro version to display loop items.

The update to Elementor Pro version 3.19.1 did not solve the problem of large gaps between the loop items or the missing stylesheet.

I have restored Elementor Pro version 3.18.3. There are no problems with this version.

**Comment by jamieburchell** [comment]
> I use the "Dynamic Template Grid" widget from "Unlimited Elements for Elementor" in the Pro version to display loop items.

If you haven't already, you should contact the authors of that plugin with your findings. They may need to make changes to make it compatible with 3.19.

**Comment by nicholaszein** [comment]
Hello everyone!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Core and Pro v3.19.1! 🥳 

✅ Feel free to check it out and update both plugins to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂

**Comment by nicholaszein** [comment]
@lachermeierm.

Please reach out to Unlimited Elements and ask them to fix your issue.
